OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -354,9 +352,10 @@ - (void)track:(NSString *)eventKey
     if ([experimentVariationMapping count] > 0) {
         userInfo[OptimizelyNotificationsUserDictionaryExperimentVariationMappingKey] = [experimentVariationMapping copy];
     }
-    [[NSNotificationCenter defaultCenter] postNotificationName:OptimizelyDidTrackEventNotification
-                                                        object:self
-                                                      userInfo:userInfo];
+    NSString *_userId = userId ? userId : @"";
+    NSDictionary *_attributes = attributes ? attributes : [NSDictionary new];
+    NSDictionary *_eventTags = eventTags ? eventTags : [NSDictionary new];
+    [_notificationCenter sendNotifications:OPTLYNotificationTypeTrack args:eventKey, _userId, _attributes, _eventTags, conversionEventParams, nil];
 }
```

Comment:
In order to implement variable arguments in sendNotifications method, I make sure no argument could be pass as nil which can break the argument list.